src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferDictionaryTryAddValueOverGuardedAddTests.cs

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Globalization;
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferDictionaryTryAddValueOverGuardedAddAnalyzer,
+    Microsoft.NetCore.CSharp.Analyzers.Performance.CSharpPreferDictionaryTryAddValueOverGuardedAddFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferDictionaryTryAddValueOverGuardedAddAnalyzer,
+    Microsoft.NetCore.VisualBasic.Analyzers.Performance.BasicPreferDictionaryTryAddValueOverGuardedAddFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class PreferDictionaryTryAddValueOverGuardedAddTests
```

Comment:
Would something like be flagged as well? 

```csharp
if (!dictionary.TryGetValue(key, out var value))
{
    dictionary.Add(key, new object());
}
// the version below allocations even if the value is already in the map
dictionary.TryAdd(key, new object());
```

In some cases, the hashing might be preferred to allocating indiscriminately.

Reply:
First of all, this analyzer flags usages of `ContainsKey`, not `TryGetValue`.

To answer your question, no, the analyzer would not flag this construct. It searches for literals only ([see here](https://github.com/CollinAlpert/roslyn-analyzers/blob/392174cdf469ef76d822e8308d8b0c75717c942b/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferDictionaryTryAddValueOverGuardedAddAnalyzer.cs#L97)), as per this requirement: https://github.com/dotnet/runtime/issues/33799#issuecomment-773499679

Reply:
Thank you!

Reply:
looking at that comment: 
> [dotnet/runtime#33799 (comment)](https://github.com/dotnet/runtime/issues/33799#issuecomment-773499679)
where the associated value is: (a) a literal; or (b) already fully hydrated before the call to ContainsKey

would be great if you could include (b) as only literals seems quite limited